.github/actions/make_init/action.yml

```diff
@@ -0,0 +1,88 @@
+name: "Setup Virtual Env"
+
+inputs:
+  use_cached_venv:
+    description: 'Use Cached Virtual Env'
+    default: 'true'
+
+runs:
+  using: composite
+  steps:
+    - name: Restore pre-commit cache
+      id: cache-pre-commit
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pre-commit
+        key: v1-pre-commit-${{ env.pythonLocation }}-${{ hashFiles('**/.pre-commit-config.yaml') }}
+    - if: steps.cache-pre-commit.outputs.cache-hit != 'true'
+      name: Install pre-commit
+      run: |
+        pip install pre-commit
```

Comment:
Shouldn't pre-commit always be installed regardless of the cache state? Now when pre-commit cache exists, pre-commit is not installed.

Reply:
Admittedly, I'm not familiar with pre-commit, and don't understand the distinction between what the pre-commit cache entails vs. installing & running `pre-commit install-hooks`. I can do some research.

Reply:
I checked and we are using pipenv pre-commit, but in this case it makes sense to install it separately.

Reply:
Sorry just want to make sure I understand - so we should have the `install pre-commit` step regardless? Or keep the `install pre-commit` step only if there is no cache hit?